sdks/common/runner-api/src/main/proto/beam_runner_api.proto

```diff
@@ -475,20 +491,53 @@ message Trigger {
 
   // The default trigger. Equivalent to Repeat { AfterEndOfWindow } but
   // specially denoted to indicate the user did not alter the triggering.
-  message Default { }
+  message Default {
+  }
+
+  // Ready whenever the requisite number of input elements have arrived
+  message ElementCount {
+    int64 element_count = 1;
+  }
 
   // The full disjoint union of possible triggers.
   oneof trigger {
     AfterAll after_all = 1;
     AfterAny after_any = 2;
     AfterEach after_each = 3;
     AfterEndOfWindow after_end_of_widow = 4;
-    AfterProcessingTimeDelay after_processing_time_delay = 5;
-    AfterSynchronizedProcessingTimeDelay after_synchronized_processing_time_delay = 6;
+    AfterProcessingTime after_processing_time = 5;
```

Comment:
Alphabetized. Since nothing uses this API, also put the indices in order. I'll stop that in a couple weeks.